mmdet/structures/bbox/base_bbox.py

```diff
@@ -0,0 +1,523 @@
+            w.r.t ``scale_facotr``. The difference is that ``resize_`` only
+            changes the width and the height of bboxes, but ``rescale_`` also
+            rescales the box centers simultaneously.
+
+        Args:
+            scale_factor (Tuple[float, float]): factors for scaling box
+                shapes. The length should be 2.
+        """
+        pass
+
+    @abstractmethod
+    def is_bboxes_inside(self, img_shape: Tuple[int, int]) -> BoolTensor:
+        """Find bboxes inside the image.
+
+        Args:
+            img_shape (Tuple[int, int]): A tuple of image height and width.
+
+        Returns:
+            BoolTensor: A BoolTensor indicating whether the box is inside
+            the image. Assuming the original boxes have shape (m, n, bbox_dim),
+            the output has shape (m, n).
+        """
+        pass
+
+    @abstractmethod
+    def find_inside_points(self,
+                           points: Tensor,
+                           is_aligned: bool = False) -> BoolTensor:
+        """Find inside box points. Boxes dimension must be 2.
+
+        Args:
+            points (Tensor): Points coordinates. Has shape of (m, 2).
+            is_aligned (bool): Whether ``points`` has been aligned with bboxes
+                or not. If True, the length of bboxes and ``points`` should be
+                the same. Defaults to False.
+
+        Returns:
+            BoolTensor: A BoolTensor indicating whether a point is inside
+            boxes. Assuming the boxes has shape of (n, bbox_dim), if
+            ``is_aligned`` is False. The index has shape of (m, n). If
+            ``is_aligned`` is True, m should be equal to n and the index has
+            shape of (m, ).
+        """
+        pass
+
+    @abstractstaticmethod
+    def bbox_overlaps(bboxes1: 'BaseBoxes',
+                      bboxes2: 'BaseBoxes',
+                      mode: str = 'iou',
+                      is_aligned: bool = False,
+                      eps: float = 1e-6) -> Tensor:
+        """Calculate overlap between two set of boxes with their modes
+        converted to the present box mode.
+
+        Args:
+            bboxes1 (:obj:`BaseBoxes`): BaseBoxes with shape of (m, bbox_dim)
+                or empty.
+            bboxes2 (:obj:`BaseBoxes`): BaseBoxes with shape of (n, bbox_dim)
+                or empty.
+            mode (str): "iou" (intersection over union), "iof" (intersection
+                over foreground). Defaults to "iou".
+            is_aligned (bool): If True, then m and n must be equal. Defaults
+                to False.
+            eps (float): A value added to the denominator for numerical
+                stability. Defaults to 1e-6.
+
+        Returns:
+            Tensor: shape (m, n) if ``is_aligned`` is False else shape (m,)
+        """
+        pass
+
+    @abstractstaticmethod
+    def from_bitmap_masks(masks: BitmapMasks) -> 'BaseBoxes':
```

Comment:
suggest unify the function from_bitmap_masks and from_polygon_masks to ``from_instance_masks``.
And handle different types inside the function. Usually it should first call `mask.to_bboxes` uniformly then create a box list.

Reply:
`mask.to_bboxes` can only generate horizontal boxes.